### PR TITLE
User or service tenant topologies

### PR DESCRIPTION
**Pull request by dougwig**

### CODE REVIEW

File: a10_neutron_lbaas/etc/defaults.py [hunk 1]
```diff
@@ -66,9 +66,9 @@
     'nova_flavor',
     'glance_image',
 
-    'vthunder_tenant_name',
-    'vthunder_tenant_username',
-    'vthunder_tenant_password',
+    # 'vthunder_tenant_name',
```
Comment (mdurrant-b3) [inline a10_neutron_lbaas/etc/defaults.py]: WTF?

**Comment by Cedev** [comment]
Can the choice of tenants be configurable instead of requiring a new plumbing hooks for each combination?

**Comment by dougwig** [comment]
Service tenant vs user tenant are two very different network topologies, which is why I liked them as separate hooks. What were you thinking, though?

**Comment by Cedev** [comment]
Service tenant vs user tenant doesn't change network topology; it changes who owns the nova instance.

I was thinking of letting the tenant that owns what vary independently in vthunder config where the tenant is currently specified. Load/make sessions for whatever those happen to be and pass 'em to instance manager.

**Comment by dougwig** [comment]
Not true, since the networks go from inside one tenant space to spanning. What you say is simple enough; if there is no service tenant defined, we default to user.  But... and it's a big but... the perms on the service tenant user are *drastically* different than an in-tenant topology, and will require pre-setup. That seems easier to document on a "separate" scheduler, doesn't it?

**Comment by mdurrant-b3** [comment]
I concur with Doug's reasoning - the differences between the two approaches and required configuration is big enough that it should be a separate scheduler.

File: a10_neutron_lbaas/vthunder/instance_manager.py [hunk 2]
```diff
@@ -347,9 +359,6 @@ def distinct_dicts(dicts):
...
 
 
+# deprecated
 def config_instance_manager(config):
-        vth = config.get_vthunder_config()
-        ks = a10_keystone.KeystoneA10(
-            config.get('keystone_version'), config.get('keystone_auth_url'), vth)
-        imgr = InstanceManager(ks_session=ks.session)
-        return imgr
+    return InstanceManager.from_config(config)
```
Comment (Cedev) [inline a10_neutron_lbaas/vthunder/instance_manager.py]: This is a minor enough change that you can nuke the deprecated method. We'll pick it up in a10-openstack.

**Comment by dougwig** [comment]
Superseded by #335

**Pull request closed without merging**